lib/pbkit/pbkit.c

```diff
@@ -3383,60 +3316,55 @@ int pb_init(void)
     //These commands assign DMA channels to push buffer subchannels
     //and associate some specific GPU parts to specific Dma channels
     p=pb_begin();
-    pb_push1to(SUBCH_2,p,NV20_TCL_PRIMITIVE_SET_MAIN_OBJECT,14); p+=2;
-    pb_push1to(SUBCH_3,p,NV20_TCL_PRIMITIVE_SET_MAIN_OBJECT,16); p+=2;
-    pb_push1to(SUBCH_4,p,NV20_TCL_PRIMITIVE_SET_MAIN_OBJECT,17); p+=2;
-    pb_push1to(SUBCH_3D,p,NV20_TCL_PRIMITIVE_SET_MAIN_OBJECT,13); p+=2;
-    pb_push1to(SUBCH_2,p,NV20_TCL_PRIMITIVE_3D_SET_OBJECT0,7); p+=2;
-    pb_push1to(SUBCH_3,p,NV20_TCL_PRIMITIVE_3D_SET_OBJECT5,17); p+=2;
-    pb_push1to(SUBCH_3,p,NV20_TCL_PRIMITIVE_3D_SET_OBJECT_UNKNOWN,3); p+=2;
-    pb_push2to(SUBCH_4,p,NV20_TCL_PRIMITIVE_3D_SET_OBJECT1,3,11); p+=3;
-    pb_end(p); //calls pb_start() which will trigger the reading and sending to GPU (asynchronous, no waiting)
+    p=pb_push1to(SUBCH_2,p,NV20_TCL_PRIMITIVE_SET_MAIN_OBJECT,14);
+    p=pb_push1to(SUBCH_3,p,NV20_TCL_PRIMITIVE_SET_MAIN_OBJECT,16);
+    p=pb_push1to(SUBCH_4,p,NV20_TCL_PRIMITIVE_SET_MAIN_OBJECT,17);
+    p=pb_push1to(SUBCH_3D,p,NV20_TCL_PRIMITIVE_SET_MAIN_OBJECT,13);
+    p=pb_push1to(SUBCH_2,p,NV20_TCL_PRIMITIVE_3D_SET_OBJECT0,7);
+    p=pb_push1to(SUBCH_3,p,NV20_TCL_PRIMITIVE_3D_SET_OBJECT5,17);
+    p=pb_push1to(SUBCH_3,p,NV20_TCL_PRIMITIVE_3D_SET_OBJECT_UNKNOWN,3);
+    p=pb_push2to(SUBCH_4,p,NV20_TCL_PRIMITIVE_3D_SET_OBJECT1,3,11);
+    p=pb_end(p); //calls pb_start() which will trigger the reading and sending to GPU (asynchronous, no waiting)
 
     //setup needed for color computations
     p=pb_begin();
-    pb_push(p++,NV20_TCL_PRIMITIVE_3D_SET_OBJECT0,3);
-    *(p++)=2;
-    *(p++)=3;
-    *(p++)=3;
+    p=pb_push3(p,NV20_TCL_PRIMITIVE_3D_SET_OBJECT0,2,3,3);
```

Comment:
Avoid this step here. The GPU will auto-increment the command index, so it's possible to send multiple commands with on header. We should handle this differently, so it's more obvious which commands are send.

I'm not sure about this particular case, but it could be up to 3 different individual commands, so we should not merge them into one line (it hides the intention). Keep the `pb_push` variant for now.

Reply:
Note: It's generally fine when only submitting 1 parameter, because then we know it's only 1 command being submitted. It's only problematic for 2 or more parameters as they *could* (and likely will) belong to different commands.

Reply:
Reverted. :+1: